pkg/core/native/policy.go

```diff
@@ -152,25 +152,20 @@ func (p *Policy) OnPersist(ic *interop.Context) error {
 
 // PostPersist implements Contract interface.
 func (p *Policy) PostPersist(ic *interop.Context) error {
-	return nil
-}
-
-// OnPersistEnd updates cached Policy values if they've been changed
-func (p *Policy) OnPersistEnd(dao dao.DAO) error {
 	if p.isValid {
 		return nil
 	}
 	p.lock.Lock()
 	defer p.lock.Unlock()
```

Comment:
Also should lock before access to `p.isValid`?

Reply:
Fixed.